[Cu f94xbn] create main layout

Buglik opened the pull request:
Main layout with two tabs( tab1 and tab2 for now) and avatar button with dropdown menu (logout option and some setting that will be provided later).

Buglik committed:
MainLayout proposition (with mobile sidebar)

Buglik committed:
better looking mobile version

Buglik committed:
Final layout version
(in desktop view logo centered)

Zummek commented:
Task linked: [CU-f94xbn Create main layout](https://app.clickup.com/t/f94xbn)

Zummek commented on web-app/src/components/HeaderMenuElement.vue:
```diff
@@ -1,14 +1,11 @@
 <template>
   <q-item clickable tag="a" target="_blank" :href="link">
-    <q-item-section v-if="icon" avatar>
+    <q-item-section v-if="icon" class="lt-md" avatar>
```
Why do You want to show icon only on lower size screen than `md`?

Buglik replied:
that's some trash from sidebar option, removing whole HeaderMenuElement anyway

Zummek commented on web-app/src/components/HeaderMenuElement.vue:
```diff
@@ -19,7 +16,6 @@ import { Vue, Component, Prop } from 'vue-property-decorator'
 @Component
 export default class EssentialLink extends Vue {
```
Change class name to this same as the file name.

Zummek commented on web-app/src/components/HeaderMenuElement.vue:
```diff
@@ -19,7 +16,6 @@ import { Vue, Component, Prop } from 'vue-property-decorator'
 @Component
 export default class EssentialLink extends Vue {
   @Prop({ type: String, required: true }) readonly title!: string
```
`label` sounds better than `title`

Zummek commented on web-app/src/layouts/MainLayout.vue:
```diff
@@ -1,41 +1,121 @@
+                  </div>
+                </div>
+                <q-separator horizontal class="q-mx-xs" />
+                <div class="row  q-pa-md">
```
```suggestion
                <div class="row q-pa-md">
```

Zummek commented on web-app/src/layouts/MainLayout.vue:
```diff
@@ -44,60 +124,26 @@
+  components: { HeaderMenuElement },
 })
 export default class MainLayout extends Vue {
   leftDrawerOpen = false;
```
It's useless variable

Buglik replied:
again some trash from sidebar version

Zummek commented on web-app/src/layouts/MainLayout.vue:
```diff
@@ -44,60 +124,26 @@
+    link: '#',
   },
   {
-    title: 'Quasar Awesome',
-    caption: 'Community Quasar projects',
-    icon: 'favorite',
-    link: 'https://awesome.quasar.dev',
+    title: 'Tab2',
+    link: '#',
```
I think we can change tab titles to `News`, `Tournaments`

Zummek reviewed (commented):
When You create/modify new view paste here screenshot

eastemployeer commented on web-app/src/layouts/MainLayout.vue:
```diff
@@ -1,41 +1,121 @@
+          />
+        </div>
+
+        <q-toolbar-title class="lt-sm">
+          <q-icon name="emoji_events" />
+          Best tournament app
+        </q-toolbar-title>
+        <q-toolbar-title class="gt-xs text-center">
+          <q-icon name="emoji_events" />
+          Best tournament app
         </q-toolbar-title>
```
```suggestion
        <q-toolbar-title :class="$q.screen.gt.xs ? 'text-center gt-xs' : ''">
          <q-icon name="emoji_events" />
          Best tournament app
        </q-toolbar-title>
```

eastemployeer commented on web-app/src/layouts/MainLayout.vue:
```diff
@@ -1,41 +1,121 @@
 
-        <div>Quasar v{{ $q.version }}</div>
+        <div class="row col-3 gt-xs justify-end">
+          <div>
```
Useless div

eastemployeer commented on web-app/src/layouts/MainLayout.vue:
```diff
@@ -1,41 +1,121 @@
-        <q-toolbar-title>
-          Quasar App
+        <div class="row col-3 gt-xs">
+          <HeaderMenuElement
```
External component for adding buttons with icons to the header? Really?

eastemployeer reviewed (changes requested):
Code can be improved and simplified in several places.

Buglik committed:
better version
simplified some of the code,
solved a problem with title overflow,
changed tabs names (also set equal width at mobile view),
deleted HeaderMenuElement,
removed some trash as well

Buglik committed:
prettier formatOnSave added

Zummek reviewed (approved):
LGTM
